docs/articles/retention.md

```diff
@@ -0,0 +1,183 @@
+
+- Top <n\> tags most recently pushed
+- Top <n\> tags most recently pulled
+- Tags pushed in the past <n\> hours
+- Tags pulled in the past <n\> hours
+- Tags matching a regular expression (regex) pattern
+
+
+## Configuring retention policies
+
+Retention policies are configured in the `storage` section of the zot configuration file under the `retention` attribute.  One or more policies can be grouped under the `policies` attribute.  
+
+By default, if no retention policies are defined, all tags are retained.
+
+> :warning: If at least one retention policy is defined, all tags not matching any policy are removed. We recommend defining a default policy to avoid unintended removals.
+
+### Configuration example
+
+The following example is a simple retention configuration with two policies:
+
+- The first includes all available configuration attributes. 
+- The second acts as a default policy.
+
+_simple policy example_
+
+```json
+  "storage": {
+    "retention": {
+      "dryRun": false,
+      "delay": "24h",
+      "policies": [
+        {
+          "repoNames": ["infra/*", "tmp/**"],
+          "deleteReferrers": false,
+          "deleteUntagged": true,
+          "KeepTags": [{
+            "patterns": ["v2.*", ".*-prod"],
+            "mostRecentlyPushedCount": 10,
+            "mostRecentlyPulledCount": 10,
+            "pulledWithin": "720h",
+            "pushedWithin": "720h"
+          }]  
+        },
+        {
+          "keepTags": [{
+            "patterns": [".*"]
+          }]
+        }
+      ]
+    }
+  }
+```
+
+### Configurable attributes
+
+The following table lists the attributes available in the retention policy configuration.
+
+| Attribute | Value | Description |
+|-----------| ----- | ------------|
+| dryRun | boolean | If `true`, will log a removal action without actually removing the image.  Default is `false`. |
+| delay | time | Remove untagged and referrers only if they are older than the specified <time\> hours, such as 24h. |
+| policies | list | A list of policies. |
+| repositories | list | A list of glob patterns to match repositories. |  
+| deleteReferrers | boolean | If true, delete manifests with a missing Subject.  Default is `false`. |
+| deleteUntagged | boolean | If true, delete untagged manifests.  Default is `true`. |
+| keepTags | list | Criteria for tags to retain always. |
+| mostRecentlyPushedCount | count | Retains the top <count\> most recently pushed tags. |
+| mostRecentlyPulledCount | count | Retains the top <count\> most recently pulled tags. |
+| pushedWithin | time | Retains the tags pushed during the last <time\> hours, such as 24h. |
+| pulledWithin | time | Retains the tags pulled during the last <time\> hours, such as 24h. |
+| patterns | regex | See Notes. |
+
+> Mike's questions:  
+> - Which is the correct attribute name: `repositories` (as in config-retention.json) or `repoNames` (as in README.md)? 
+> - Are these statements conflicting? 
```

Comment:
no,

if a repo matches then it will proceed to check its tag retention policies and applies them.
if a repo doesn't match then retention is not applied.

- If at least one keepTags policy is defined for a repository, all tags not matching those policies are removed[fixed]
- If a repository doesn't match any retention policies, all its tags are retained.